src/sentry/models/organization.py

```diff
@@ -188,12 +191,19 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         if not self.id:
-            self.id = snowflake_id_generation()
+            self.id = snowflake_id_generation(self.snowflake_redis_key)
         if not self.slug:
             lock = locks.get("slug:organization", duration=5)
             with TimedRetryPolicy(10)(lock.acquire):
                 slugify_instance(self, self.name, reserved=RESERVED_ORGANIZATION_SLUGS)
-        super().save(*args, **kwargs)
+        try:
+            with transaction.atomic():
+                super().save(*args, **kwargs)
+        except IntegrityError:
+            if self.snowflake_retry_counter == 5:
```

Comment:
How does this counter get incremented?

Reply:
oops will fix for both this and the case below